src/envelope.rs

```diff
@@ -1,6 +1,7 @@
 use std::{fmt::Display, net::SocketAddr};
 
 use bytes::Bytes;
+use futures::channel::oneshot;
```

Comment:
can [Tokio's oneshot channel](https://docs.rs/tokio/latest/tokio/sync/oneshot/index.html) be used instead instead? it doesn't depend on the tokio runtime being active, so if you're worried about circular dependencies, it's not an issue here.

Reply:
Oops, messed up the import. Tokio is what we want.

Reply:
whoops, might've commented on a stable revision. sorry about that!